python/lsst/ip/diffim/zogy.py

```diff
@@ -898,3 +949,133 @@ class ZogyMapReduceConfig(ImageMapReduceConfig):
+            mi -= mn2[0]
+
+        self.log.info('Running Zogy algorithm: spatiallyVarying=%r' % spatiallyVarying)
+
+        if not self._validateWcs(templateExposure, scienceExposure):
+            if doWarping:
+                self.log.info("Astrometrically registering template to science image")
+                # Also warp the PSF
+                xyTransform = afwImage.XYTransformFromWcsPair(scienceExposure.getWcs(),
+                                                              templateExposure.getWcs())
+                psfWarped = measAlg.WarpedPsf(templateExposure.getPsf(), xyTransform)
+                templateExposure = self._warper.warpExposure(scienceExposure.getWcs(),
+                                                             templateExposure,
+                                                             destBBox=scienceExposure.getBBox())
+
+                templateExposure.setPsf(psfWarped)
+                templateExposure.writeFits('WARPEDTEMPLATE_ZOGY.fits')
+            else:
+                self.log.error("ERROR: Input images not registered")
+                raise RuntimeError("Input images not registered")
+
+        def gm(exp):
+            return exp.getMaskedImage().getMask()
+        def ga(exp):
+            return exp.getMaskedImage().getImage().getArray()
+        def gv(exp):
+            return exp.getMaskedImage().getImage().getArray()
+
+        if spatiallyVarying:
+            config = self.config.zogyMapReduceConfig
+            task = ImageMapReduceTask(config=config)
+            results = task.run(scienceExposure, template=templateExposure, inImageSpace=inImageSpace,
+                               doScorr=doPreConvolve, forceEvenSized=True)
+            results.D = results.exposure
+            # The CoaddPsf apparently cannot be used for detection as it doesn't have a
+            #  getImage() or computeShape() method (which uses getAveragePosition(), which apparently
+            #  is not implemented correctly.
+            # Need to get it to return the matchedExposure (convolved template) too, for dipole fitting.
+        else:
+            config = self.config.zogyConfig
+            task = ZogyTask(scienceExposure=scienceExposure, templateExposure=templateExposure,
+                            config=config)
+            if not doPreConvolve:
+                results = task.computeDiffim(inImageSpace=inImageSpace)
+                results.matchedExposure = results.R
+            else:
+                results = task.computeScorr(inImageSpace=inImageSpace)
+                results.D = results.S
+
+        # Make sure masks of input images are propagated to diffim
+        mask = results.D.getMaskedImage().getMask()
+        badBits = mask.getPlaneBitMask(['UNMASKEDNAN', 'NO_DATA', 'BAD', 'EDGE', 'SUSPECT', 'CR', 'SAT'])
+        badBitsNan = mask.getPlaneBitMask(['UNMASKEDNAN'])
+        mask |= gm(scienceExposure)
+        mask |= gm(templateExposure)
+        gm(results.D)[:, :] = mask
```

Comment:
`mask` here is a view (pointer) of `results.D`'s mask  (i.e. the two lines before have already modified it)  This line is unnecessary as it just copies itself to itself.